e2e/rook-test-framework/rook-infra-manager/rook_test_infra_manager.go

```diff
@@ -0,0 +1,323 @@
+package rook_infra_manager
+
+import (
+	"bytes"
+	"errors"
+	"fmt"
+	"github.com/rook/rook/e2e/rook-test-framework/contracts"
+	"github.com/rook/rook/e2e/rook-test-framework/enums"
+	"github.com/rook/rook/e2e/rook-test-framework/objects"
+	"github.com/rook/rook/e2e/rook-test-framework/transport"
+	"github.com/rook/rook/e2e/rook-test-framework/utils"
+	"io/ioutil"
+	"strings"
+	"sync"
+	"time"
+	"runtime"
+	"path"
+)
+
+type rookTestInfraManager struct {
+	transportClient contracts.ITransportClient
+	platformType    enums.RookPlatformType
+	dockerized      bool
+	dockerContext   *objects.DockerContext
+	k8sVersion      enums.K8sVersion
+}
+
+var (
+	r    *rookTestInfraManager
+	once sync.Once
+	curdir string
+)
+
+const (
+	rookOperatorFileName = "rook-operator.yaml"
+	rookClusterFileName = "rook-cluster.yaml"
+	rookClientFileName = "rook-client.yml"
+	podSpecPath = "../../../demo/kubernetes"
+	scriptsPath = "scripts"
+	k8sFalsePostiveSuccessErrorMsg = "exit status 1"
```

Comment:
this message is always a false positive for failure? that's surprising, I would at least add some comments about why that is or an example operation you'd see this from.